src/linklabel.rs

```diff
@@ -35,6 +40,15 @@ pub(crate) fn scan_link_label<'a>(text: &'a str) -> Option<(usize, LinkLabel<'a>
     let mut byte_index = 1;
     // no worries, doesnt allocate until we push things onto it
     let mut label = String::new();
+    let is_footnote = if let Some(&'^') = char_iter.peek() {
```

Comment:
Footnotes are optional (they will cause divergence with the CommonMark spec), so I'd really like to see the option plumbed through.